src/offline/cable_parameters.F90

```diff
@@ -2064,7 +2064,195 @@ SUBROUTINE check_parameter_values(soil, veg, ssnow)
     END WHERE
 
   END SUBROUTINE check_parameter_values
+  
+  !===============================================================================
+
+  SUBROUTINE consistency_ice_veg_soil(soil, veg)  
+    ! Ensure that when an active patch has a veg type of ice then its soil type is also ice and vice versa
+    ! Any change effected to enforce this consistency includes correcting the appropriate paramter values 
+
+    USE grid_constants_mod_cbl, ONLY : ICE_SoilType, ICE_VegType
+    USE cable_phys_constants_mod, ONLY : csice, density_ice
+    
+    TYPE (soil_parameter_type), INTENT(INOUT) :: soil  ! soil parameter data
+    TYPE (veg_parameter_type),  INTENT(INOUT) :: veg   ! vegetation parameter
+
+    INTEGER :: i, j, k, kIVP
+    LOGICAL :: LIceVegPatch
+
+    
+    LIceVegPatch = .FALSE.
+
+    look_for_ice_veg_patch:     DO i = 1, mland                 ! loop over all gridcells
+       DO j = 1, landpt(i)%nap      ! loop over all active patches in every gridcell
+          
+          k = landpt(i)%cstart + j - 1    ! absolute position of active veg patch 
+          
+          IF(veg%iveg(k) == ICE_VegType) THEN  ! check to see if there is at least one ice veg patch already
+             LIceVegPatch = .TRUE.
+             kIVP = k    ! remember which patch it is 
+             EXIT look_for_ice_veg_patch ! exit as we only need to find one ice veg patch 
+          END IF
+          
+       END DO
+    END DO look_for_ice_veg_patch
+
+    
+    DO i = 1, mland                 ! loop over all gridcells 
+       DO j = 1, landpt(i)%nap      ! loop over all active patches in every gridcell 
+          
+          k = landpt(i)%cstart + j - 1    ! absolute position of active patch 
+          
+          IF(soil%isoilm(k) == ICE_SoilType) THEN  ! check to see if ice soil patch
+
+             WRITE(*,*) 'SUBROUTINE load_parameters:'
+             WRITE(*,*) 'At land point number ', i
+             WRITE(*,*) 'And patch number ', k
+             WRITE(*,*) 'isoilm is ICE_SoilType'
+             WRITE(*,*) 'Set rhosoil = density_ice from CABLE physical constants'
+             WRITE(*,*) 'Set css     = csice from CABLE physical constants'
```

Comment:
The write statements need to be directed to `logn` logical unit so that they are written in the CABLE log file and not the standard output. 
`logn` is in `cable_IO_vars_module` which is in the USE statements on this `cable_param_module` module so it shouldn't need to be passed in by argument.

This is a comment valid for all the WRITE() statements not just the ones here.

Reply:
@ccarouge Happy to make the change but just want to point out before I do so that I modelled these write statements after the ones in SUBROUTINE check_parameter_values. Overall there are 41 occurrences of WRITE(*,*) in cable_parameters.F90. I am guessing then that you are planning eventually to change all of them ?

Reply:
Ok, I made the requested changes.

Reply:
We will work on existing stuff later. For the moment, let's make sure the new stuff gives the info in the correct place. Thanks for making the change.